### DISCUSSION TITLE
How to add a passphrase to an ed25519 private key?

### DISCUSSION THREAD

**Discussion by mohitatray** [paulmillr/noble-ed25519#107] [Q&A] [answered]
Apologies for this silly question. But can you tell me how I can protect the private key with a passphrase. Should I use AES-GCM to encrypt the private key? How do I derive the AES-GCM key with a passphrase?

**Comment by paulmillr** [comment, marked as answer]
You can use PBKDF2 or argon2id to derive aes-gcm key from passphrase. they are available in noble-hashes
